subcommand/acl-controller/command.go

```diff
@@ -213,44 +212,40 @@ func (c *Command) upsertConsulResources(consulClient *api.Client, ecsMeta awsuti
 
 	agentSelf, err := consulClient.Agent().Self()
 	if err != nil {
-		return fmt.Errorf("failed to get Consul agent self config: %w", err)
+		return fmt.Errorf("failed to get Consul server agent's self config: %w", err)
 	}
 	var agentConfig AgentConfig
 	err = mapstructure.Decode(agentSelf, &agentConfig)
 	if err != nil {
-		return fmt.Errorf("failed to decode Consul agent self config: %w", err)
+		return fmt.Errorf("failed to decode Consul server agent's self config: %w", err)
 	}
 
-	if c.flagPartitionsEnabled {
-		if c.flagPartition == "" {
+	if c.config.Controller.PartitionsEnabled {
+		if c.config.Controller.Partition == "" {
 			// if an explicit partition was not provided use the default partition.
-			c.flagPartition = controller.DefaultPartition
+			c.config.Controller.Partition = controller.DefaultPartition
 		}
 		if err := c.upsertPartition(consulClient); err != nil {
 			return err
 		}
-	} else if c.flagPartition != "" {
-		return fmt.Errorf("partition flag provided without partitions-enabled flag")
+	} else if c.config.Controller.Partition != "" {
+		return fmt.Errorf("`config.controller.partition` provided without setting `config.controller.partitionEnabled` as true")
 	}
 
-	if err := c.upsertConsulClientRole(consulClient, roleName, policyName); err != nil {
-		return err
-	}
-	if err := c.upsertAuthMethod(consulClient, clientAuthMethod); err != nil {
-		return err
-	}
 	if err := c.upsertAuthMethod(consulClient, serviceAuthMethod); err != nil {
 		return err
 	}
-	if err := c.upsertBindingRule(consulClient, clientBindingRule); err != nil {
-		return err
-	}
 	if err := c.upsertBindingRule(consulClient, serviceBindingRule); err != nil {
 		return err
 	}
 	if err := c.upsertAnonymousTokenPolicy(consulClient, agentConfig); err != nil {
 		return err
 	}
+
+	if err := c.registerNode(consulClient, ecsMeta, clusterARN); err != nil {
+		return fmt.Errorf("registering ecs cluster as a node in Consul: %s", err)
```

Comment:
nit: Use `%w` for error wrapping:

```suggestion
		return fmt.Errorf("registering ECS cluster as a node in Consul: %w", err)
```